src/client/common/variables/environmentVariablesProvider.ts

```diff
@@ -57,22 +56,6 @@ export class EnvironmentVariablesProvider implements IEnvironmentVariablesProvid
         promise.then((result) => (cacheStoreIndexedByWorkspaceFolder.data = result)).ignoreErrors();
         return promise;
     }
-    public async _getEnvironmentVariables(resource?: Uri): Promise<EnvironmentVariables> {
-        let mergedVars = await this.getCustomEnvironmentVariables(resource);
-        if (!mergedVars) {
-            mergedVars = {};
-        }
-        this.envVarsService.mergeVariables(mergedVars!, process.env);
```

Comment:
The problem was in a previous PR i flipped these two arguments, that was wrong.

Reply:
Yeah, I remember this function from a previous review. Target is the second arg, so this would have merged it into process.env. 

I don't think that we have to change this now, but given that this function has caused some issues maybe it's a problem? Might make more sense as a side effect free version.

mergeVariables(readonly source: EnvVars, readonly target: EnvVars): EnvVars.

Then:

mergedVars = mergeVariables(process.env, mergedVars);

Don't need to change now I think. But the current version does require digging into the function to see what is actually happening.